src/Searching.ts

```diff
@@ -125,7 +125,7 @@ async function combinedSearch(searchTerm: string): Promise<ISearchResults> {
     const emptyResult: ISeshatSearchResults = {
         seshatQuery: localQuery,
         _query: serverQuery,
-        serverSideNextBatch: serverResponse.next_batch,
+        serverSideNextBatch: serverResponse.search_categories.room_events.next_batch,
```

Comment:
Hmm, I remember testing this extensively but this would mean that it would fetch a server response only once.

It does seem legit, perhaps I changed  something in the last refactoring pass which might not have been tested as extensively.